### PR TITLE
add delete and rename for time

### PR DESCRIPTION
**Pull request by VladVega**
1. delete `pt[timeField]` after putting the value into `pt.time`. 
2. if there exists a `pt.time` with a separate `timeField`, avoid simply discarding by setting that value to key `pt._time`.  

(also note that I will have to do a little rework on this after https://github.com/juttle/juttle/pull/248 since the function moves to a new class - but the function does not change in logic in that pr)

This PR originated from @dmajda comment here: https://github.com/juttle/juttle-sql-adapter-common/pull/16

And has to do with bringing this back: https://github.com/juttle/juttle-sql-adapter-common/pull/14

### CODE REVIEW

File: lib/runtime/procs/source.js [hunk 1]
```diff
@@ -22,17 +22,21 @@ var source = base.extend({
...
         _.each(points, function(pt) {
-            if (timeField && !pt[timeField]) {
-                self.trigger('warning', self.runtime_error('RT-POINT-MISSING-TIME', {field: timeField}));
+            if (timeField) {
+                if (!pt[timeField]) {
+                    self.trigger('warning', self.runtime_error('RT-POINT-MISSING-TIME', {field: timeField}));
+                }
+                if (timeField !== 'time') {
+                    if (pt.time) {
+                        pt._time = pt.time;
```
Comment (demmer) [inline lib/runtime/procs/source.js]: In talking this over with @dmajda this morning, we agreed that having this "magic" behavior was undesirable.

Instead the logic should be that if there is a field called `time` in the point and the `timeField` is not time (either undefined or something else), then we should throw a warning and remove `time` from the point.

To support the case where the user wants to keep that field but stash it in something else, then we should add a `-timeRenameField` option which would implement the logic to move the conflicting time into another field. I don't love the name of that option FWIW but I think it's the least bad that I could think of.

**Comment by VladVega** [comment]
@demmer what do you thing about this solution?

**Comment by VladVega** [comment]
@davidvgalbraith also.

**Comment by VladVega** [comment]
Notes from @demmer:
1. Consider toNative + parseTime => same function.
2. Ensure point is dropped with error when `time = foo`.

**Pull request closed without merging**